tensorlayer/layers/convolution.py

```diff
@@ -1153,6 +1154,82 @@ def __init__(
             self.all_params.append(filters)
 
 
+class AtrousConv2dTransLayer(Layer):
+    """The :class:`AtrousConv2dTransLayer` class is 2D atrous convolution transpose, see `tf.nn.atrous_conv2d_transpose <https://www.tensorflow.org/versions/master/api_docs/python/nn.html#atrous_conv2d_transpose>`__.
+
+    Parameters
+    ----------
+    prev_layer : :class:`Layer`
+        Previous layer with a 4D output tensor in the shape of (batch, height, width, channels).
+    shape : tuple of int
+        The shape of the filters: (filter_height, filter_width, out_channels, in_channels).
+    output_shape : tuple of int
+        Output shape of the deconvolution.
+    rate : int
+        The stride that we sample input values in the height and width dimensions.
+        This equals the rate that we up-sample the filters by inserting zeros across the height and width dimensions.
+        In the literature, this parameter is sometimes mentioned as input stride or dilation.
+    act : activation function
+        The activation function of this layer.
+    padding : str
+        The padding algorithm type: "SAME" or "VALID".
+    W_init : initializer
+        The initializer for the weight matrix.
+    b_init : initializer or None
+        The initializer for the bias vector. If None, skip biases.
+    W_init_args : dictionary
+        The arguments for the weight matrix initializer.
+    b_init_args : dictionary
+        The arguments for the bias vector initializer.
+    name : str
+        A unique layer name.
+
+    """
+
+    @deprecated_alias(layer='prev_layer', end_support_version=1.9)  # TODO remove this line for the 1.9 release
+    def __init__(
+            self, prev_layer, shape=(3, 3, 128, 256), output_shape=(1, 64, 64, 128), rate=2, act=tf.identity,
```

Comment:
Please format with one argument per line

Reply:
Because `rate` can be adjusted differently for the same `output_shape`, it is not redundant information.

Reply:
@DEKHTIARJonathan The current `yapf` configuration does not match your recommendation. Please update `setup.cfg` [link](https://github.com/tensorlayer/tensorlayer/blob/master/setup.cfg).